uidesignjobapp/components/example/signup-form-demo.tsx

```diff
@@ -1,99 +1,411 @@
-          <Label htmlFor="email">Email Address</Label>
-          <Input id="email" placeholder="projectmayhem@fc.com" type="email" />
+          <Label htmlFor="email">Email Address <span className="text-red-500">*</span></Label>
+          <Input id="email" placeholder="projectmayhem@fc.com" type="email" 
+          onChange={(e) => setFormData({ ...formData, email: e.target.value })}
+          onBlur={(e) => validateEmail(e.target.value)} />
+          {errors.email && <p className="text-red-500 text-xs">{errors.email}</p>}
         </LabelInputContainer>
         <LabelInputContainer className="mb-4">
-          <Label htmlFor="password">Password</Label>
-          <Input id="password" placeholder="••••••••" type="password" />
-        </LabelInputContainer>
-        <LabelInputContainer className="mb-8">
-          <Label htmlFor="twitterpassword">Your twitter password</Label>
-          <Input
-            id="twitterpassword"
-            placeholder="••••••••"
-            type="twitterpassword"
-          />
+          <Label htmlFor="phoneNumber">Mobile Phone Number <span className="text-red-500">*</span></Label>
+          <Input id="phoneNumber" placeholder="(000) 000-0000" type="phoneNumber" value={formData.phoneNumber} 
+          onChange={(e) => handleInputChange("phoneNumber", e.target.value)}
+          onBlur={(e) => handleInputChange("phoneNumber", e.target.value)} />
+          {errors.phoneNumber && <p className="text-red-500 text-xs">{errors.phoneNumber}</p>}
         </LabelInputContainer>
 
-        <button
-          className="bg-gradient-to-br relative group/btn from-black dark:from-zinc-900 dark:to-zinc-900 to-neutral-600 block dark:bg-zinc-800 w-full text-white rounded-md h-10 font-medium shadow-[0px_1px_0px_0px_#ffffff40_inset,0px_-1px_0px_0px_#ffffff40_inset] dark:shadow-[0px_1px_0px_0px_var(--zinc-800)_inset,0px_-1px_0px_0px_var(--zinc-800)_inset]"
-          type="submit"
-        >
-          Sign up &rarr;
-          <BottomGradient />
-        </button>
+        <div className="mb-8">
+          <LabelInputContainer className="mb-4">
+            <Label htmlFor="street">Street</Label>
+            <Input id="street" placeholder="123 Paper St" type="text" />
+          </LabelInputContainer>
+          <div className="flex space-x-2">
+            <LabelInputContainer>
+              <Label htmlFor="city">City</Label>
+              <Input id="city" placeholder="Wilmington" type="text" />
+            </LabelInputContainer>
+            <LabelInputContainer>
+              <Label htmlFor="zip">Zip Code</Label>
+              <Input id="zip" placeholder="19801" type="text" />
+            </LabelInputContainer>
+          </div>
+          <LabelInputContainer className="mt-4">
+            <Label htmlFor="state">State</Label>
+            <select id="state" className="border border-gray-300 rounded px-3 py-2">
+              <option value="">Select</option>
+              <option value="AL">Alabama</option>
```

Comment:
Totally fine for this class imo but there's probably some library we can use if this was the senior project class for example so we don't have to write so many constants.